### PR TITLE
User friendly docs for operations

### LINKED ISSUE #898: Nice operations pages in docs
### Desired Behaviour

Help links in operations should show a user friendly page, instead of API reference.

### PR DESCRIPTION
**Pull request by samtygier-stfc**
### Issue

Closes #898

(note the number is the branch name is wrong)

### Description

Add extension to generate user friendly documentation for operations. Takes the docstring for the class, and the parameters from the `filter_func` method (with some standard parameters removed).

Update doc strings for operations

Also: Move test skipping into decorators to avoid errors with sphinx

### Testing & Acceptance Criteria 

To build the docs locally run:

`python3 ./setup.py docs`

Docs will be written to docs/build/html , so for me operations docs are in ~/git/mantidimaging/docs/build/html/user_guide/operations/index.html

Help button in the operations window should link to something like https://mantidproject.github.io/mantidimaging/user_guide/operations/index.html#crop-coordinates

### Documentation
Updated release notes

### CODE REVIEW

File: mantidimaging/core/operations/remove_dead_stripe/remove_dead_stripe.py
Comment (StephenNneji) [inline mantidimaging/core/operations/remove_dead_stripe/remove_dead_stripe.py:18]: What is axis 0? Will the user know what this is?

File: mantidimaging/core/operations/circular_mask/circular_mask.py
Comment (StephenNneji) [inline mantidimaging/core/operations/circular_mask/circular_mask.py:19]: Do you mean the edge or border instead of  "on the outside"?

File: mantidimaging/core/operations/rescale/rescale.py
Comment (StephenNneji) [inline mantidimaging/core/operations/rescale/rescale.py:22]: Might be better to say:
When: Automatically used when saving as 16 bit images

File: mantidimaging/core/operations/clip_values/clip_values.py
Comment (StephenNneji) [inline mantidimaging/core/operations/clip_values/clip_values.py:12]: I think **pixel values** is clearer than **grey values** but I will leave that to you. Just ensure consistency, it should be _grey_ or _gray_

File: mantidimaging/core/operations/roi_normalisation/roi_normalisation.py
Comment (StephenNneji) [inline mantidimaging/core/operations/roi_normalisation/roi_normalisation.py:59]: **grayvalues** needs space

File: mantidimaging/core/operations/rotate_stack/rotate_stack.py
Comment (StephenNneji) [inline mantidimaging/core/operations/rotate_stack/rotate_stack.py:26]: small tilts **are** taken into account at the reconstruction stage

File: mantidimaging/core/operations/divide/divide.py
Comment (StephenNneji) [inline mantidimaging/core/operations/divide/divide.py:17]: Is this dividing the pixel values?

File: mantidimaging/core/operations/remove_dead_stripe/remove_dead_stripe.py
Comment (samtygier-stfc) [reply mantidimaging/core/operations/remove_dead_stripe/remove_dead_stripe.py:18]: I'll remove the "axis 0" sentence. Its from the sarepy docs, and useful for developers not MI users.

File: mantidimaging/core/operations/circular_mask/circular_mask.py
Comment (samtygier-stfc) [reply mantidimaging/core/operations/circular_mask/circular_mask.py:19]: Changed to "on the outer edge of the image." to be clearer.

File: mantidimaging/core/operations/rescale/rescale.py
Comment (samtygier-stfc) [reply mantidimaging/core/operations/rescale/rescale.py:22]: Changed to "Automatically used when saving as unsigned integer image formats, to avoid clipping negative values".

Its the unsigned bit that matters.

**Review by StephenNneji — COMMENTED** [review]
The operations page looks significantly better which should improve user experience 👍. I will approve after comments are addressed.

File: mantidimaging/core/operations/rescale/rescale.py
Comment (StephenNneji) [reply mantidimaging/core/operations/rescale/rescale.py:22]: I thought image formats are always unsigned?

File: mantidimaging/core/operations/clip_values/clip_values.py
Comment (samtygier-stfc) [reply mantidimaging/core/operations/clip_values/clip_values.py:12]: Winfried suggested "grey value", so I'll go with that.

File: mantidimaging/core/operations/rotate_stack/rotate_stack.py
Comment (samtygier-stfc) [reply mantidimaging/core/operations/rotate_stack/rotate_stack.py:26]: I'll go with "can be taken", because its not automatic

File: mantidimaging/core/operations/divide/divide.py
Comment (samtygier-stfc) [reply mantidimaging/core/operations/divide/divide.py:17]: I think this is to scale the attenuation values by the pixel size. I think maybe this should be hidden from the operations window, as it is called automatically after reconstruction.

Comment (StephenNneji) [reply mantidimaging/core/operations/divide/divide.py:17]: Yes, it hiding it may prevent confusion later on

**Review by StephenNneji — APPROVED** [review]

File: mantidimaging/core/operations/rescale/rescale.py
Comment (samtygier-stfc) [reply mantidimaging/core/operations/rescale/rescale.py]: Pretty sure that FITS and TIFF can be signed (though we currently only write uint16). Lots of graphics software wont open non uint tiffs.

**Pull request merged**
